### PR TITLE
Automate website updates with netlify

### PR DESCRIPTION
**Pull request by stp-ip**
On each PR it will build the site and publish it on a review url.
This will also pull in any changes from the last tag within coredns/coredns.

When merged aka a production deploy is triggered via netlify. This enables an additional step. It will create an automatic commit to push back any changes that where pulled in from coredns/coredns.

That way we automatically pull in changes from coredns/coredns and auto deploy to coredns.netlify.com

If this seems feasible in general. Happy to work out the kinks:
* [ ] change DNS of coredns.io to `coredns.io IN A 104.198.14.52`
* [ ] use deploy key instead of oauth token for push
* [ ] corecheck isn't run due to failures

Another option to write the data back into the repo would be to push to the PR instead.
That way the specific merge is still reviewed.
Happy to discuss this, if necessary.

### CODE REVIEW

File: bin/sync-from-coredns.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/python
 
-# Copy the <plugin>/README.md to plugins/<mddleware>.md, add
+# Copy the <plugin>/README.md to plugins/<middleware>.md, add
```
Comment (johnbelamaric) [inline bin/sync-from-coredns.py]: shouldn't that be `<plugin>.md`?

File: content/plugins/file.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ file DBFILE [ZONES... ] {
 Load the `example.org` zone from `example.org.signed` and allow transfers to the internet, but send
 notifies to 10.240.1.1
 
-~~~ corefile
```
Comment (johnbelamaric) [inline content/plugins/file.md]: these are used to do some automated testing, I believe. is this change necessary?

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+coredns
+public
```
Comment (johnbelamaric) [inline .gitignore]: missing EOL

**Comment by stp-ip** [comment]
In general processwise:

Netlify is the provider
CNCF pays the bills and also has access to the account
Control of the account currently includes 3 people (2xCNCF and me)
Logs are currently public

Deploy is mostly automatic:
* Create PR
* Merge PR
* Netlify deploy will pull and commit any changes (from CoreDNS/CoreDNS) to
master, while deploying the site


Will work out the other issues as well next week.

File: content/plugins/file.md (hunk 2 above)
Comment (stp-ip) [reply content/plugins/file.md]: That one is auto synced and therefore a merged thing within coredns/coredns.

Comment (stp-ip) [reply content/plugins/file.md]: Should be fixed with coredns/coredns#3051 and the next automatic run with version >1.5.2

**Comment by stp-ip** [comment]
Will take another stab at this in the coming weeks.

**Comment by miekg** [comment]
Excellent. Thank you.

**Comment by stp-ip** [comment]
Will be replaced by a few PRs coming. Closing for now
